File: recette/src/main/java/com/example/recette/bean/Ingredient.java
package com.example.recette.bean;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Ingredient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String ref;
    private String nom;
    private int quantite;
    private String unite;
}

File: front-end/src/app/models/ingredient.ts
export class Ingredient {
  id: number;
  ref: string;
  nom: string;
  quantite: number;
  unite: string;

  constructor(id: number,   ref: string,    nom: string, quantite: number, unite: string) {
    this.ref = ref;
    this.id = id;
    this.nom = nom;
    this.quantite = quantite;
    this.unite = unite;
  }
}
